airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/jsonschema/JsonSchemaIdentityMapper.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.integrations.destination.s3.jsonschema
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import io.airbyte.commons.jackson.MoreMappers
+
+open class JsonSchemaIdentityMapper : JsonSchemaMapper() {
+    override fun mapObjectWithProperties(schema: ObjectNode): ObjectNode {
+        val newSchema = MoreMappers.initMapper().createObjectNode()
+        val newProperties = MoreMappers.initMapper().createObjectNode()
+
+        newSchema.put("type", "object")
+        schema["properties"].fields().forEach { (key, value) ->
+            newProperties.set<ObjectNode>(key, mapSchema(value as ObjectNode))
+        }
+        newSchema.replace("properties", newProperties)
+
+        return newSchema
+    }
+
+    override fun mapObjectWithoutProperties(schema: ObjectNode): ObjectNode {
```

Comment:
I chose to denormalize the types as much as possible, however I agree that was arbitrary. It works either way. I was going for the smallest possible methods/max user flexibility. It was more useful for things like `combined`/`union` and `array[With/Without]Items`, because it changed the access pattern for their elements.

I think when I change this to use our bespoke airbyte schema, the json -> bespoke mapping should merge into the smallest number of types possible, with a common interface. (`getProperties/getOptions/etc`)

Reply:
makes sense (I forgot that accessing jsonschema as JsonNode is annoying). And +1 on merging to the smallest number of methods once we're on AirbyteType (I think we can get away with just 4 methods for struct, array, union, primitive at that point)